src/pymatgen/cli/pmg_config.py

```diff
@@ -202,8 +202,8 @@ def build_bader(fortran_command="gfortran"):
     cwd = os.getcwd()
     state = True
     try:
-        urlretrieve(bader_url, "bader.tar.gz")
-        subprocess.call(["tar", "-zxf", "bader.tar.gz"])
+        urlretrieve(bader_url, "bader.tar.gz")  # noqa: S310
+        subprocess.call(["/usr/bin/tar", "-zxf", "bader.tar.gz"])
```

Comment:
not sure we can hard-code `/usr/bin/tar` here. leaving it at `tar` will search everything on the user's path, not just 1 directory

Reply:
I was quite concern about these changes as well, https://github.com/materialsproject/pymatgen/pull/3892#discussion_r1650911892, as security is not a core concern for `pymatgen`. Explicitly use the absolute path could avoid PATH hijacking.

Reply:
@janosh I'm still pretty concerned about commit 26ccb8591b3363783b15f224c6da09311f81d5cc which includes using the absolute path for the following:
- /usr/bin/gunzip
- /usr/bin/gzip
- /usr/bin/make
- /usr/bin/tar
- /bin/cp

There are pretty basic packages but owing to the huge diversities of Linux distro, I'm unsure if there would be any exception. According to ChatGPT 4o:

> the Filesystem Hierarchy Standard (FHS) provides guidelines and standards for directory structures and file placements in Unix-like operating systems, including Linux. The FHS is maintained by the Linux Foundation.

>Here are some relevant points from the FHS regarding the locations of these binaries:

>    /bin: This directory contains essential command binaries that are required for single-user mode and for all users (e.g., cp, ls, cat).

>    /usr/bin: This directory contains the majority of user commands. Binaries that are not essential for system boot or repair but are intended for use by all users are placed here. This includes binaries like gzip, gunzip, make, and tar.

It should be quite safe IMO.